pkgs/development/compilers/julia/1.6.nix

```diff
@@ -0,0 +1,164 @@
+
+  dontUseCmakeConfigure = true;
+  enableParallelBuilding = true;
+
+  buildInputs = [blas lapack libunwind utf8proc]
+                ++ lib.optionals stdenv.isDarwin [CoreServices ApplicationServices];
+
+  nativeBuildInputs = [cmake curl gfortran m4 makeWrapper patchelf perl python3 which];
+
+  makeFlags =
+    let
+      arch = head (splitString "-" stdenv.system);
+      march = {
+        x86_64 = stdenv.hostPlatform.gcc.arch or "x86-64";
+        i686 = "pentium4";
+        aarch64 = "armv8-a";
+      }.${arch}
+              or (throw "unsupported architecture: ${arch}");
+      # Julia requires Pentium 4 (SSE2) or better
+      cpuTarget = { x86_64 = "x86-64"; i686 = "pentium4"; aarch64 = "generic"; }.${arch}
+                  or (throw "unsupported architecture: ${arch}");
+    # Julia applies a lot of patches to its dependencies, so for now do not use the system LLVM
+    # https://github.com/JuliaLang/julia/tree/master/deps/patches
+    in [
+      "ARCH=${arch}"
+      "MARCH=${march}"
+      "JULIA_CPU_TARGET=${cpuTarget}"
+      "PREFIX=$(out)"
+      "prefix=$(out)"
+      "SHELL=${stdenv.shell}"
+
+      "USE_SYSTEM_BLAS=1"
+      "USE_BLAS64=${if blas.isILP64 then "1" else "0"}"
+
+      "USE_SYSTEM_LAPACK=1"
+
+      "USE_SYSTEM_ARPACK=1"
+      "USE_SYSTEM_FFTW=1"
+      "USE_SYSTEM_GMP=1"
+      "USE_SYSTEM_LIBGIT2=1"
+      "USE_SYSTEM_LIBUNWIND=1"
+
+      "USE_SYSTEM_MPFR=1"
+      "USE_SYSTEM_OPENLIBM=1"
+      "USE_SYSTEM_OPENSPECFUN=1"
+      "USE_SYSTEM_PATCHELF=1"
+      "USE_SYSTEM_PCRE=1"
+      "PCRE_CONFIG=${pcre2.dev}/bin/pcre2-config"
+      "PCRE_INCL_PATH=${pcre2.dev}/include/pcre2.h"
+      "USE_SYSTEM_READLINE=1"
+      "USE_SYSTEM_UTF8PROC=1"
+      "USE_SYSTEM_ZLIB=1"
+
+      "USE_BINARYBUILDER=0"
+    ];
+
+  LD_LIBRARY_PATH = makeLibraryPath [
+    arpack
+    blas
+    curl
+    fftw
+    fftwSinglePrec
+    gmp
+    lapack
+    libgit2
+    libnghttp2
+    libunwind
+    mpfr
+    openlibm
+    openspecfun
+    pcre2
+    utf8proc
+  ];
+
+  # Julia's tests require read/write access to $HOME
+  preCheck = ''
+    export HOME="$NIX_BUILD_TOP"
+  '';
+
+  preBuild = ''
+    sed -e '/^install:/s@[^ ]*/doc/[^ ]*@@' -i Makefile
+    sed -e '/[$](DESTDIR)[$](docdir)/d' -i Makefile
+    export LD_LIBRARY_PATH=${LD_LIBRARY_PATH}
+  '';
+
+  postInstall = ''
+    # Symlink shared libraries from LD_LIBRARY_PATH into lib/julia,
+    # as using a wrapper with LD_LIBRARY_PATH causes segmentation
+    # faults when program returns an error:
+    #   $ julia -e 'throw(Error())'
+    find $(echo $LD_LIBRARY_PATH | sed 's|:| |g') -maxdepth 1 -name '*.${if stdenv.isDarwin then "dylib" else "so"}*' | while read lib; do
+      if [[ ! -e $out/lib/julia/$(basename $lib) ]]; then
+        ln -sv $lib $out/lib/julia/$(basename $lib)
+      fi
+    done
+  '';
+
+  passthru = {
+    inherit majorVersion minorVersion maintenanceVersion;
+    site = "share/julia/site/v${majorVersion}.${minorVersion}";
+  };
+
+  meta = {
```

Comment:
```suggestion
  meta = with lib; {
```

Reply:
I believe we actually have a `with lib;` all the way up in line 16: https://github.com/NixOS/nixpkgs/pull/118000/files?file-filters%5B%5D=.nix&file-filters%5B%5D=.py#diff-108b1b5f124b298e85e1ddd222047f1acdbb4d87916e0ffd051637d16ccc261fR16.

Let me take a look at cleaning this up a bit...